src/main/java/com/conveyal/datatools/manager/jobs/DeployJob.java

```diff
@@ -228,7 +228,11 @@ public void jobLogic () {
             // Dump the deployment bundle to the temp file.
             try {
                 status.message = "Creating transit bundle (GTFS and OSM)";
-                this.deployment.dump(deploymentTempFile, true, true, true);
+                // Only download OSM extract if an OSM extract does not exist at a public URL and not skipping extract.
+                boolean includeOsm = deployment.osmExtractUrl == null && !deployment.skipOsmExtract;
+                // TODO: At this stage, perform a HEAD request on OSM extract URL to verify that it exists before
```

Comment:
I am not sure where the existence is checked. I don't see it here or inside `deployment.dump()`